### ISSUE TITLE
CSS utility frameworks

### ISSUE THREAD

**Issue by oliverfoster** [adaptlearning/adapt_framework#3283]
### Subject of the enhancement
Include a CSS utility framework. 

References:
* https://css-tricks.com/need-css-utility-library/
* https://github.com/troxler/awesome-css-frameworks
* https://tailwindcss.com/

**Comment by paulstevendev** [comment]
We had a look at tailwind css for another projectand we were not keen on how much additional mark up it adds so we scrapped the idea of using it. This is not to say it may not work for Adapt.

**Comment by StuartNicholls** [comment]
I tend to agree that including something like tailwind may not be that helpful *but* building on what we already have, and creating more adapt specific utility classes for various things could really help out with modernisation, things like apis for buttons and layout stuff, and integrating with AAT would be hugely beneficial.
